zookeeper/zookeeper.php

```diff
@@ -0,0 +1,398 @@
+<?php
+
+/**
+ * Zookeeper class.
+ * @link https://www.php.net/manual/en/class.zookeeper.php
+ */
+class Zookeeper
+{
+    /* class constants */
+    const PERM_READ   = 1;
+    const PERM_WRITE  = 2;
+    const PERM_CREATE = 4;
+    const PERM_DELETE = 8;
+    const PERM_ADMIN  = 16;
+    const PERM_ALL    = 31;
+
+    const EPHEMERAL = 1;
+    const SEQUENCE  = 2;
+
+    const EXPIRED_SESSION_STATE = -112;
+    const AUTH_FAILED_STATE     = -113;
+    const CONNECTING_STATE      = 1;
+    const ASSOCIATING_STATE     = 2;
+    const CONNECTED_STATE       = 3;
+    const NOTCONNECTED_STATE    = 999;
+
+    const CREATED_EVENT     = 1;
+    const DELETED_EVENT     = 2;
+    const CHANGED_EVENT     = 3;
+    const CHILD_EVENT       = 4;
+    const SESSION_EVENT     = -1;
+    const NOTWATCHING_EVENT = -2;
+
+    const LOG_LEVEL_ERROR = 1;
+    const LOG_LEVEL_WARN  = 2;
+    const LOG_LEVEL_INFO  = 3;
+    const LOG_LEVEL_DEBUG = 4;
+
+    const SYSTEMERROR          = -1;
+    const RUNTIMEINCONSISTENCY = -2;
+    const DATAINCONSISTENCY    = -3;
+    const CONNECTIONLOSS       = -4;
+    const MARSHALLINGERROR     = -5;
+    const UNIMPLEMENTED        = -6;
+    const OPERATIONTIMEOUT     = -7;
+    const BADARGUMENTS         = -8;
+    const INVALIDSTATE         = -9;
+
+    const OK                      = 0;
+    const APIERROR                = -100;
+    const NONODE                  = -101;
+    const NOAUTH                  = -102;
+    const BADVERSION              = -103;
+    const NOCHILDRENFOREPHEMERALS = -108;
+    const NODEEXISTS              = -110;
+    const NOTEMPTY                = -111;
+    const SESSIONEXPIRED          = -112;
+    const INVALIDCALLBACK         = -113;
+    const INVALIDACL              = -114;
+    const AUTHFAILED              = -115;
+    const CLOSING                 = -116;
+    const NOTHING                 = -117;
+    const SESSIONMOVED            = -118;
+
+    /**
+     * Create a handle to used communicate with zookeeper.
+     * If the host is provided, attempt to connect.
+     *
+     * @param string   $host
+     * @param callable $watcher_cb
+     * @param int      $recv_timeout
+     *
+     * @link https://www.php.net/manual/en/zookeeper.construct.php
+     *
+     * @throws ZookeeperConnectionException when host is provided and when failed to connect to the host
+     */
+    public function __construct($host = '', $watcher_cb = null, $recv_timeout = 10000)
```

Comment:
This method calls `php_zookeeper_connect_impl`, which may call `php_zk_throw_exception(ZBADARGUMENTS TSRMLS_CC)` which throws `ZookeeperException`.